三级菜单添加后用户端无法访问halo内容网页

### What is version of Halo has the issue?

2.0.0的alpha.2

### What database are you using?

H2

### What is your deployment method?

Fat Jar

### Your site address.

公网demo环境地址

### What happened?

添加三级子菜单后，如下图所示三级菜单添加完成后，

<img width="893" alt="image" src="https://user-images.githubusercontent.com/60059766/195543531-7d52374e-54c6-45c9-a4a5-7417a8223b39.png">

会导致用户端无法访问网站，提示错误如下：

Whitelabel Error Page
This application has no configured error view, so you are seeing this as a fallback.

Thu Oct 13 15:33:23 CST 2022
[81153ee4-1112] There was an unexpected error (type=Internal Server Error, status=500).




### Relevant log output

_No response_

### Additional information

_No response_

观测到的日志：

```
2022-10-14T17:30:58.433+08:00 ERROR 7 --- [undedElastic-60] org.thymeleaf.TemplateEngine             : [THYMELEAF][boundedElastic-60] Exception processing template "index": Exception evaluating SpringEL expression: "menuFinder.getDefault()" (template: "modules/header" - line 4, col 5)

org.thymeleaf.exceptions.TemplateProcessingException: Exception evaluating SpringEL expression: "menuFinder.getDefault()" (template: "modules/header" - line 4, col 5)
	at org.thymeleaf.spring6.expression.SPELVariableExpressionEvaluator.evaluate(SPELVariableExpressionEvaluator.java:292) ~[thymeleaf-spring6-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.standard.expression.VariableExpression.executeVariableExpression(VariableExpression.java:166) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.standard.expression.SimpleExpression.executeSimple(SimpleExpression.java:66) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.standard.expression.Expression.execute(Expression.java:109) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.standard.expression.Expression.execute(Expression.java:138) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.standard.expression.Expression.execute(Expression.java:125) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.standard.processor.StandardWithTagProcessor.doProcess(StandardWithTagProcessor.java:93) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.processor.element.AbstractAttributeTagProcessor.doProcess(AbstractAttributeTagProcessor.java:74) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.processor.element.AbstractElementTagProcessor.process(AbstractElementTagProcessor.java:95) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.util.ProcessorConfigurationUtils$ElementTagProcessorWrapper.process(ProcessorConfigurationUtils.java:633) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.ProcessorTemplateHandler.handleOpenElement(ProcessorTemplateHandler.java:1314) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.OpenElementTag.beHandled(OpenElementTag.java:205) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.Model.process(Model.java:282) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.ProcessorTemplateHandler.handleStandaloneElement(ProcessorTemplateHandler.java:1204) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.StandaloneElementTag.beHandled(StandaloneElementTag.java:228) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.Model.process(Model.java:282) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.ProcessorTemplateHandler.handleOpenElement(ProcessorTemplateHandler.java:1587) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.OpenElementTag.beHandled(OpenElementTag.java:205) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.TemplateModel.process(TemplateModel.java:136) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.TemplateManager.parseAndProcess(TemplateManager.java:592) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1103) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1077) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at run.halo.app.theme.engine.SpringWebFluxTemplateEngine.lambda$createFullStream$0(SpringWebFluxTemplateEngine.java:202) ~[classes/:2.0.0-alpha.2]
	at reactor.core.publisher.MonoCreate.subscribe(MonoCreate.java:58) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
	at reactor.core.publisher.Mono.subscribe(Mono.java:4379) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
	at reactor.core.publisher.MonoSubscribeOn$SubscribeOnSubscriber.run(MonoSubscribeOn.java:126) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[na:na]
	at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
Caused by: java.lang.NullPointerException: Cannot invoke "run.halo.app.theme.finders.vo.MenuItemVo.setParentName(String)" because "childrenVo" is null
	at run.halo.app.theme.finders.impl.MenuFinderImpl.lambda$populateParentName$9(MenuFinderImpl.java:123) ~[classes/:2.0.0-alpha.2]
	at java.base/java.lang.Iterable.forEach(Unknown Source) ~[na:na]
	at run.halo.app.theme.finders.impl.MenuFinderImpl.lambda$populateParentName$10(MenuFinderImpl.java:121) ~[classes/:2.0.0-alpha.2]
	at java.base/java.util.ArrayList.forEach(Unknown Source) ~[na:na]
	at run.halo.app.theme.finders.impl.MenuFinderImpl.populateParentName(MenuFinderImpl.java:114) ~[classes/:2.0.0-alpha.2]
	at run.halo.app.theme.finders.impl.MenuFinderImpl.listAsTree(MenuFinderImpl.java:63) ~[classes/:2.0.0-alpha.2]
	at run.halo.app.theme.finders.impl.MenuFinderImpl.getDefault(MenuFinderImpl.java:45) ~[classes/:2.0.0-alpha.2]
	at jdk.internal.reflect.GeneratedMethodAccessor385.invoke(Unknown Source) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Unknown Source) ~[na:na]
	at org.springframework.expression.spel.support.ReflectiveMethodExecutor.execute(ReflectiveMethodExecutor.java:139) ~[spring-expression-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.expression.spel.ast.MethodReference.getValueInternal(MethodReference.java:112) ~[spring-expression-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.expression.spel.ast.MethodReference$MethodValueRef.getValue(MethodReference.java:386) ~[spring-expression-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.expression.spel.ast.CompoundExpression.getValueInternal(CompoundExpression.java:92) ~[spring-expression-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.expression.spel.ast.SpelNodeImpl.getValue(SpelNodeImpl.java:112) ~[spring-expression-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.expression.spel.standard.SpelExpression.getValue(SpelExpression.java:338) ~[spring-expression-6.0.0-M6.jar:6.0.0-M6]
	at org.thymeleaf.spring6.expression.SPELVariableExpressionEvaluator.evaluate(SPELVariableExpressionEvaluator.java:265) ~[thymeleaf-spring6-3.1.0.M2.jar:3.1.0.M2]
	... 32 common frames omitted

2022-10-14T17:30:58.434+08:00 ERROR 7 --- [undedElastic-60] r.h.a.t.e.SpringWebFluxTemplateEngine    : [THYMELEAF][boundedElastic-60] Exception processing template "index": Exception evaluating SpringEL expression: "menuFinder.getDefault()" (template: "modules/header" - line 4, col 5)

org.thymeleaf.exceptions.TemplateProcessingException: Exception evaluating SpringEL expression: "menuFinder.getDefault()" (template: "modules/header" - line 4, col 5)
	at org.thymeleaf.spring6.expression.SPELVariableExpressionEvaluator.evaluate(SPELVariableExpressionEvaluator.java:292) ~[thymeleaf-spring6-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.standard.expression.VariableExpression.executeVariableExpression(VariableExpression.java:166) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.standard.expression.SimpleExpression.executeSimple(SimpleExpression.java:66) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.standard.expression.Expression.execute(Expression.java:109) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.standard.expression.Expression.execute(Expression.java:138) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.standard.expression.Expression.execute(Expression.java:125) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.standard.processor.StandardWithTagProcessor.doProcess(StandardWithTagProcessor.java:93) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.processor.element.AbstractAttributeTagProcessor.doProcess(AbstractAttributeTagProcessor.java:74) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.processor.element.AbstractElementTagProcessor.process(AbstractElementTagProcessor.java:95) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.util.ProcessorConfigurationUtils$ElementTagProcessorWrapper.process(ProcessorConfigurationUtils.java:633) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.ProcessorTemplateHandler.handleOpenElement(ProcessorTemplateHandler.java:1314) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.OpenElementTag.beHandled(OpenElementTag.java:205) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.Model.process(Model.java:282) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.ProcessorTemplateHandler.handleStandaloneElement(ProcessorTemplateHandler.java:1204) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.StandaloneElementTag.beHandled(StandaloneElementTag.java:228) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.Model.process(Model.java:282) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.ProcessorTemplateHandler.handleOpenElement(ProcessorTemplateHandler.java:1587) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.OpenElementTag.beHandled(OpenElementTag.java:205) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.TemplateModel.process(TemplateModel.java:136) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.TemplateManager.parseAndProcess(TemplateManager.java:592) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1103) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1077) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at run.halo.app.theme.engine.SpringWebFluxTemplateEngine.lambda$createFullStream$0(SpringWebFluxTemplateEngine.java:202) ~[classes/:2.0.0-alpha.2]
	at reactor.core.publisher.MonoCreate.subscribe(MonoCreate.java:58) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
	at reactor.core.publisher.Mono.subscribe(Mono.java:4379) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
	at reactor.core.publisher.MonoSubscribeOn$SubscribeOnSubscriber.run(MonoSubscribeOn.java:126) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[na:na]
	at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
Caused by: java.lang.NullPointerException: Cannot invoke "run.halo.app.theme.finders.vo.MenuItemVo.setParentName(String)" because "childrenVo" is null
	at run.halo.app.theme.finders.impl.MenuFinderImpl.lambda$populateParentName$9(MenuFinderImpl.java:123) ~[classes/:2.0.0-alpha.2]
	at java.base/java.lang.Iterable.forEach(Unknown Source) ~[na:na]
	at run.halo.app.theme.finders.impl.MenuFinderImpl.lambda$populateParentName$10(MenuFinderImpl.java:121) ~[classes/:2.0.0-alpha.2]
	at java.base/java.util.ArrayList.forEach(Unknown Source) ~[na:na]
	at run.halo.app.theme.finders.impl.MenuFinderImpl.populateParentName(MenuFinderImpl.java:114) ~[classes/:2.0.0-alpha.2]
	at run.halo.app.theme.finders.impl.MenuFinderImpl.listAsTree(MenuFinderImpl.java:63) ~[classes/:2.0.0-alpha.2]
	at run.halo.app.theme.finders.impl.MenuFinderImpl.getDefault(MenuFinderImpl.java:45) ~[classes/:2.0.0-alpha.2]
	at jdk.internal.reflect.GeneratedMethodAccessor385.invoke(Unknown Source) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Unknown Source) ~[na:na]
	at org.springframework.expression.spel.support.ReflectiveMethodExecutor.execute(ReflectiveMethodExecutor.java:139) ~[spring-expression-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.expression.spel.ast.MethodReference.getValueInternal(MethodReference.java:112) ~[spring-expression-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.expression.spel.ast.MethodReference$MethodValueRef.getValue(MethodReference.java:386) ~[spring-expression-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.expression.spel.ast.CompoundExpression.getValueInternal(CompoundExpression.java:92) ~[spring-expression-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.expression.spel.ast.SpelNodeImpl.getValue(SpelNodeImpl.java:112) ~[spring-expression-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.expression.spel.standard.SpelExpression.getValue(SpelExpression.java:338) ~[spring-expression-6.0.0-M6.jar:6.0.0-M6]
	at org.thymeleaf.spring6.expression.SPELVariableExpressionEvaluator.evaluate(SPELVariableExpressionEvaluator.java:265) ~[thymeleaf-spring6-3.1.0.M2.jar:3.1.0.M2]
	... 32 common frames omitted

2022-10-14T17:30:58.437+08:00 ERROR 7 --- [undedElastic-60] a.w.r.e.AbstractErrorWebExceptionHandler : [9b2a5a3c-4120]  500 Server Error for HTTP GET "/"

org.thymeleaf.exceptions.TemplateProcessingException: Exception evaluating SpringEL expression: "menuFinder.getDefault()" (template: "modules/header" - line 4, col 5)
	at org.thymeleaf.spring6.expression.SPELVariableExpressionEvaluator.evaluate(SPELVariableExpressionEvaluator.java:292) ~[thymeleaf-spring6-3.1.0.M2.jar:3.1.0.M2]
	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException:
Error has been observed at the following site(s):
	*__checkpoint ⇢ Handler run.halo.app.theme.router.strategy.IndexRouteStrategy$$Lambda$1685/0x0000000801501ce0@663cea2a [DispatcherHandler]
	*__checkpoint ⇢ org.springframework.security.web.server.WebFilterChainProxy [DefaultWebFilterChain]
	*__checkpoint ⇢ org.springframework.boot.actuate.metrics.web.reactive.server.MetricsWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ HTTP GET "/" [ExceptionHandlingWebHandler]
Original Stack Trace:
		at org.thymeleaf.spring6.expression.SPELVariableExpressionEvaluator.evaluate(SPELVariableExpressionEvaluator.java:292) ~[thymeleaf-spring6-3.1.0.M2.jar:3.1.0.M2]
		at org.thymeleaf.standard.expression.VariableExpression.executeVariableExpression(VariableExpression.java:166) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
		at org.thymeleaf.standard.expression.SimpleExpression.executeSimple(SimpleExpression.java:66) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
		at org.thymeleaf.standard.expression.Expression.execute(Expression.java:109) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
		at org.thymeleaf.standard.expression.Expression.execute(Expression.java:138) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
		at org.thymeleaf.standard.expression.Expression.execute(Expression.java:125) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
		at org.thymeleaf.standard.processor.StandardWithTagProcessor.doProcess(StandardWithTagProcessor.java:93) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
		at org.thymeleaf.processor.element.AbstractAttributeTagProcessor.doProcess(AbstractAttributeTagProcessor.java:74) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
		at org.thymeleaf.processor.element.AbstractElementTagProcessor.process(AbstractElementTagProcessor.java:95) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
		at org.thymeleaf.util.ProcessorConfigurationUtils$ElementTagProcessorWrapper.process(ProcessorConfigurationUtils.java:633) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
		at org.thymeleaf.engine.ProcessorTemplateHandler.handleOpenElement(ProcessorTemplateHandler.java:1314) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
		at org.thymeleaf.engine.OpenElementTag.beHandled(OpenElementTag.java:205) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
		at org.thymeleaf.engine.Model.process(Model.java:282) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
		at org.thymeleaf.engine.ProcessorTemplateHandler.handleStandaloneElement(ProcessorTemplateHandler.java:1204) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
		at org.thymeleaf.engine.StandaloneElementTag.beHandled(StandaloneElementTag.java:228) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
		at org.thymeleaf.engine.Model.process(Model.java:282) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
		at org.thymeleaf.engine.ProcessorTemplateHandler.handleOpenElement(ProcessorTemplateHandler.java:1587) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
		at org.thymeleaf.engine.OpenElementTag.beHandled(OpenElementTag.java:205) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
		at org.thymeleaf.engine.TemplateModel.process(TemplateModel.java:136) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
		at org.thymeleaf.engine.TemplateManager.parseAndProcess(TemplateManager.java:592) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
		at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1103) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
		at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1077) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
		at run.halo.app.theme.engine.SpringWebFluxTemplateEngine.lambda$createFullStream$0(SpringWebFluxTemplateEngine.java:202) ~[classes/:2.0.0-alpha.2]
		at reactor.core.publisher.MonoCreate.subscribe(MonoCreate.java:58) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
		at reactor.core.publisher.Mono.subscribe(Mono.java:4379) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
		at reactor.core.publisher.MonoSubscribeOn$SubscribeOnSubscriber.run(MonoSubscribeOn.java:126) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
		at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
		at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
		at java.base/java.util.concurrent.FutureTask.run(Unknown Source) ~[na:na]
		at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) ~[na:na]
		at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[na:na]
		at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[na:na]
		at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
Caused by: java.lang.NullPointerException: Cannot invoke "run.halo.app.theme.finders.vo.MenuItemVo.setParentName(String)" because "childrenVo" is null
	at run.halo.app.theme.finders.impl.MenuFinderImpl.lambda$populateParentName$9(MenuFinderImpl.java:123) ~[classes/:2.0.0-alpha.2]
	at java.base/java.lang.Iterable.forEach(Unknown Source) ~[na:na]
	at run.halo.app.theme.finders.impl.MenuFinderImpl.lambda$populateParentName$10(MenuFinderImpl.java:121) ~[classes/:2.0.0-alpha.2]
	at java.base/java.util.ArrayList.forEach(Unknown Source) ~[na:na]
	at run.halo.app.theme.finders.impl.MenuFinderImpl.populateParentName(MenuFinderImpl.java:114) ~[classes/:2.0.0-alpha.2]
	at run.halo.app.theme.finders.impl.MenuFinderImpl.listAsTree(MenuFinderImpl.java:63) ~[classes/:2.0.0-alpha.2]
	at run.halo.app.theme.finders.impl.MenuFinderImpl.getDefault(MenuFinderImpl.java:45) ~[classes/:2.0.0-alpha.2]
	at jdk.internal.reflect.GeneratedMethodAccessor385.invoke(Unknown Source) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Unknown Source) ~[na:na]
	at org.springframework.expression.spel.support.ReflectiveMethodExecutor.execute(ReflectiveMethodExecutor.java:139) ~[spring-expression-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.expression.spel.ast.MethodReference.getValueInternal(MethodReference.java:112) ~[spring-expression-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.expression.spel.ast.MethodReference$MethodValueRef.getValue(MethodReference.java:386) ~[spring-expression-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.expression.spel.ast.CompoundExpression.getValueInternal(CompoundExpression.java:92) ~[spring-expression-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.expression.spel.ast.SpelNodeImpl.getValue(SpelNodeImpl.java:112) ~[spring-expression-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.expression.spel.standard.SpelExpression.getValue(SpelExpression.java:338) ~[spring-expression-6.0.0-M6.jar:6.0.0-M6]
	at org.thymeleaf.spring6.expression.SPELVariableExpressionEvaluator.evaluate(SPELVariableExpressionEvaluator.java:265) ~[thymeleaf-spring6-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.standard.expression.VariableExpression.executeVariableExpression(VariableExpression.java:166) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.standard.expression.SimpleExpression.executeSimple(SimpleExpression.java:66) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.standard.expression.Expression.execute(Expression.java:109) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.standard.expression.Expression.execute(Expression.java:138) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.standard.expression.Expression.execute(Expression.java:125) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.standard.processor.StandardWithTagProcessor.doProcess(StandardWithTagProcessor.java:93) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.processor.element.AbstractAttributeTagProcessor.doProcess(AbstractAttributeTagProcessor.java:74) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.processor.element.AbstractElementTagProcessor.process(AbstractElementTagProcessor.java:95) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.util.ProcessorConfigurationUtils$ElementTagProcessorWrapper.process(ProcessorConfigurationUtils.java:633) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.ProcessorTemplateHandler.handleOpenElement(ProcessorTemplateHandler.java:1314) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.OpenElementTag.beHandled(OpenElementTag.java:205) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.Model.process(Model.java:282) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.ProcessorTemplateHandler.handleStandaloneElement(ProcessorTemplateHandler.java:1204) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.StandaloneElementTag.beHandled(StandaloneElementTag.java:228) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.Model.process(Model.java:282) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.ProcessorTemplateHandler.handleOpenElement(ProcessorTemplateHandler.java:1587) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.OpenElementTag.beHandled(OpenElementTag.java:205) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.TemplateModel.process(TemplateModel.java:136) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.engine.TemplateManager.parseAndProcess(TemplateManager.java:592) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1103) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1077) ~[thymeleaf-3.1.0.M2.jar:3.1.0.M2]
	at run.halo.app.theme.engine.SpringWebFluxTemplateEngine.lambda$createFullStream$0(SpringWebFluxTemplateEngine.java:202) ~[classes/:2.0.0-alpha.2]
	at reactor.core.publisher.MonoCreate.subscribe(MonoCreate.java:58) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
	at reactor.core.publisher.Mono.subscribe(Mono.java:4379) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
	at reactor.core.publisher.MonoSubscribeOn$SubscribeOnSubscriber.run(MonoSubscribeOn.java:126) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[na:na]
	at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]

2022-10-14T17:31:00.590+08:00 ERROR 7 --- [or-http-epoll-2] a.w.r.e.AbstractErrorWebExceptionHandler : [3cad3a72-4122]  500 Server Error for HTTP GET "/apis/api.console.halo.run/v1alpha1/posts?keyword=&size=20&page=1"

java.lang.ClassCastException: null
```

--- a/src/main/java/run/halo/app/theme/finders/impl/MenuFinderImpl.java
+++ b/src/main/java/run/halo/app/theme/finders/impl/MenuFinderImpl.java
@@ -1,7 +1,8 @@
 package run.halo.app.theme.finders.impl;
 
+import java.time.Instant;
+import java.util.Collection;
 import java.util.Comparator;
-import java.util.HashMap;
 import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Map;
@@ -9,6 +10,7 @@ import java.util.Objects;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 import org.springframework.util.CollectionUtils;
+import org.springframework.util.comparator.Comparators;
 import run.halo.app.core.extension.Menu;
 import run.halo.app.core.extension.MenuItem;
 import run.halo.app.extension.ReactiveExtensionClient;
@@ -51,7 +53,6 @@ public class MenuFinderImpl implements MenuFinder {
         return menuVos.get(0);
     }
 
-
     List<MenuVo> listAll() {
         return client.list(Menu.class, null, null)
             .map(MenuVo::from)
@@ -60,7 +61,7 @@ public class MenuFinderImpl implements MenuFinder {
     }
 
     List<MenuVo> listAsTree() {
-        List<MenuItemVo> menuItemVos = populateParentName(listAllMenuItem());
+        Collection<MenuItemVo> menuItemVos = populateParentName(listAllMenuItem());
         List<MenuItemVo> treeList = listToTree(menuItemVos);
         Map<String, MenuItemVo> nameItemRootNodeMap = treeList.stream()
             .collect(Collectors.toMap(item -> item.getMetadata().getName(), Function.identity()));
@@ -74,62 +75,65 @@ public class MenuFinderImpl implements MenuFinder {
                 List<MenuItemVo> menuItems = menuItemNames.stream()
                     .map(nameItemRootNodeMap::get)
                     .filter(Objects::nonNull)
+                    .sorted(defaultTreeNodeComparator())
                     .toList();
                 return menuVo.withMenuItems(menuItems);
             })
             .toList();
     }
 
-    static List<MenuItemVo> listToTree(List<MenuItemVo> list) {
-        Map<String, List<MenuItemVo>> nameIdentityMap = list.stream()
-            .filter(item -> item.getParentName() != null)
+    static List<MenuItemVo> listToTree(Collection<MenuItemVo> list) {
+        Map<String, List<MenuItemVo>> parentNameIdentityMap = list.stream()
+            .filter(menuItemVo -> menuItemVo.getParentName() != null)
             .collect(Collectors.groupingBy(MenuItemVo::getParentName));
-        list.forEach(node -> node.setChildren(nameIdentityMap.get(node.getMetadata().getName())));
-        // clear map to release memory
-        nameIdentityMap.clear();
+
+        list.forEach(node -> {
+            // sort children
+            List<MenuItemVo> children =
+                parentNameIdentityMap.getOrDefault(node.getMetadata().getName(), List.of())
+                    .stream()
+                    .sorted(defaultTreeNodeComparator())
+                    .toList();
+            node.setChildren(children);
+        });
+
         return list.stream()
             .filter(v -> v.getParentName() == null)
             .collect(Collectors.toList());
     }
 
     List<MenuItemVo> listAllMenuItem() {
-        Function<MenuItem, Integer> priority = menuItem -> menuItem.getSpec().getPriority();
-        Function<MenuItem, String> name = menuItem -> menuItem.getMetadata().getName();
-
-        return client.list(MenuItem.class, null,
-                Comparator.comparing(priority).thenComparing(name).reversed()
-            )
+        return client.list(MenuItem.class, null, null)
             .map(MenuItemVo::from)
             .collectList()
             .block();
     }
 
-    static List<MenuItemVo> populateParentName(List<MenuItemVo> menuItemVos) {
+    static Comparator<MenuItemVo> defaultTreeNodeComparator() {
+        Function<MenuItemVo, Integer> priority = menuItem -> menuItem.getSpec().getPriority();
+        Function<MenuItemVo, Instant> createTime = menuItem -> menuItem.getMetadata()
+            .getCreationTimestamp();
+        Function<MenuItemVo, String> name = menuItem -> menuItem.getMetadata().getName();
+
+        return Comparator.comparing(priority)
+            .thenComparing(createTime, Comparators.nullsLow())
+            .thenComparing(name);
+    }
+
+    static Collection<MenuItemVo> populateParentName(List<MenuItemVo> menuItemVos) {
         Map<String, MenuItemVo> nameIdentityMap = menuItemVos.stream()
             .collect(Collectors.toMap(menuItem -> menuItem.getMetadata().getName(),
                 Function.identity()));
 
-        Map<String, MenuItemVo> treeVoMap = new HashMap<>();
-        // populate parentName
-        menuItemVos.forEach(menuItemVo -> {
-            final String parentName = menuItemVo.getMetadata().getName();
-            treeVoMap.putIfAbsent(parentName, menuItemVo);
-            LinkedHashSet<String> children = menuItemVo.getSpec().getChildren();
-            if (CollectionUtils.isEmpty(children)) {
-                return;
+        nameIdentityMap.forEach((name, value) -> {
+            LinkedHashSet<String> children = value.getSpec().getChildren();
+            if (children != null) {
+                for (String child : children) {
+                    MenuItemVo childNode = nameIdentityMap.get(child);
+                    childNode.setParentName(name);
+                }
             }
-            children.forEach(childrenName -> {
-                MenuItemVo childrenVo = nameIdentityMap.get(childrenName);
-                childrenVo.setParentName(parentName);
-                treeVoMap.putIfAbsent(childrenVo.getMetadata().getName(), childrenVo);
-            });
         });
-        // clear map to release memory
-        nameIdentityMap.clear();
-        Function<MenuItemVo, Integer> priorityCmp = menuItem -> menuItem.getSpec().getPriority();
-        return treeVoMap.values()
-            .stream()
-            .sorted(Comparator.comparing(priorityCmp))
-            .toList();
+        return nameIdentityMap.values();
     }
 }

--- a/src/main/java/run/halo/app/theme/finders/vo/MenuItemVo.java
+++ b/src/main/java/run/halo/app/theme/finders/vo/MenuItemVo.java
@@ -1,6 +1,5 @@
 package run.halo.app.theme.finders.vo;
 
-import java.util.Iterator;
 import java.util.List;
 import lombok.Builder;
 import lombok.Data;
@@ -18,7 +17,7 @@ import run.halo.app.extension.MetadataOperator;
 @Data
 @ToString
 @Builder
-public class MenuItemVo {
+public class MenuItemVo implements VisualizableTreeNode<MenuItemVo> {
 
     MetadataOperator metadata;
 
@@ -56,20 +55,8 @@ public class MenuItemVo {
             .build();
     }
 
-    void print(StringBuilder buffer, String prefix, String childrenPrefix) {
-        buffer.append(prefix);
-        buffer.append(getDisplayName());
-        buffer.append('\n');
-        if (children == null) {
-            return;
-        }
-        for (Iterator<MenuItemVo> it = children.iterator(); it.hasNext(); ) {
-            MenuItemVo next = it.next();
-            if (it.hasNext()) {
-                next.print(buffer, childrenPrefix + "├── ", childrenPrefix + "│   ");
-            } else {
-                next.print(buffer, childrenPrefix + "└── ", childrenPrefix + "    ");
-            }
-        }
+    @Override
+    public String nodeText() {
+        return getDisplayName();
     }
 }

--- a/src/main/java/run/halo/app/theme/finders/vo/MenuItemVo.java
+++ b/src/main/java/run/halo/app/theme/finders/vo/MenuItemVo.java
@@ -1,9 +1,11 @@
 package run.halo.app.theme.finders.vo;
 
+import java.util.Iterator;
 import java.util.List;
 import lombok.Builder;
 import lombok.Data;
 import lombok.ToString;
+import org.apache.commons.lang3.StringUtils;
 import run.halo.app.core.extension.MenuItem;
 import run.halo.app.extension.MetadataOperator;
 
@@ -29,6 +31,16 @@ public class MenuItemVo {
     String parentName;
 
     /**
+     * Gets menu item's display name.
+     */
+    public String getDisplayName() {
+        if (status != null && StringUtils.isNotBlank(status.getDisplayName())) {
+            return status.getDisplayName();
+        }
+        return spec.getDisplayName();
+    }
+
+    /**
      * Convert {@link MenuItem} to {@link MenuItemVo}.
      *
      * @param menuItem menu item extension
@@ -42,5 +54,22 @@ public class MenuItemVo {
             .status(status)
             .children(List.of())
             .build();
+    }
+
+    void print(StringBuilder buffer, String prefix, String childrenPrefix) {
+        buffer.append(prefix);
+        buffer.append(getDisplayName());
+        buffer.append('\n');
+        if (children == null) {
+            return;
+        }
+        for (Iterator<MenuItemVo> it = children.iterator(); it.hasNext(); ) {
+            MenuItemVo next = it.next();
+            if (it.hasNext()) {
+                next.print(buffer, childrenPrefix + "├── ", childrenPrefix + "│   ");
+            } else {
+                next.print(buffer, childrenPrefix + "└── ", childrenPrefix + "    ");
+            }
+        }
     }
 }

--- a/src/main/java/run/halo/app/theme/finders/vo/MenuVo.java
+++ b/src/main/java/run/halo/app/theme/finders/vo/MenuVo.java
@@ -1,5 +1,6 @@
 package run.halo.app.theme.finders.vo;
 
+import java.util.Iterator;
 import java.util.List;
 import lombok.Builder;
 import lombok.ToString;
@@ -38,5 +39,21 @@ public class MenuVo {
             .spec(menu.getSpec())
             .menuItems(List.of())
             .build();
+    }
+
+    public void print(StringBuilder buffer) {
+        buffer.append(getSpec().getDisplayName());
+        buffer.append('\n');
+        if (menuItems == null) {
+            return;
+        }
+        for (Iterator<MenuItemVo> it = menuItems.iterator(); it.hasNext(); ) {
+            MenuItemVo next = it.next();
+            if (it.hasNext()) {
+                next.print(buffer, "├── ", "│   ");
+            } else {
+                next.print(buffer, "└── ", "    ");
+            }
+        }
     }
 }

--- a/src/main/java/run/halo/app/theme/finders/vo/VisualizableTreeNode.java
+++ b/src/main/java/run/halo/app/theme/finders/vo/VisualizableTreeNode.java
@@ -1,0 +1,37 @@
+package run.halo.app.theme.finders.vo;
+
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Show Tree Hierarchy.
+ *
+ * @author guqing
+ * @since 2.0.0
+ */
+public interface VisualizableTreeNode<T extends VisualizableTreeNode<T>> {
+
+    /**
+     * Visualize tree node.
+     */
+    default void print(StringBuilder buffer, String prefix, String childrenPrefix) {
+        buffer.append(prefix);
+        buffer.append(nodeText());
+        buffer.append('\n');
+        if (getChildren() == null) {
+            return;
+        }
+        for (Iterator<T> it = getChildren().iterator(); it.hasNext(); ) {
+            T next = it.next();
+            if (it.hasNext()) {
+                next.print(buffer, childrenPrefix + "├── ", childrenPrefix + "│   ");
+            } else {
+                next.print(buffer, childrenPrefix + "└── ", childrenPrefix + "    ");
+            }
+        }
+    }
+
+    String nodeText();
+
+    List<T> getChildren();
+}

--- a/src/test/java/run/halo/app/theme/finders/impl/MenuFinderImplTest.java
+++ b/src/test/java/run/halo/app/theme/finders/impl/MenuFinderImplTest.java
@@ -1,19 +1,18 @@
 package run.halo.app.theme.finders.impl;
 
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.ArgumentMatchers.eq;
 
 import java.util.Collections;
 import java.util.LinkedHashSet;
 import java.util.List;
-import org.json.JSONException;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.Mock;
 import org.mockito.Mockito;
 import org.mockito.junit.jupiter.MockitoExtension;
-import org.skyscreamer.jsonassert.JSONAssert;
 import reactor.core.publisher.Flux;
 import reactor.util.function.Tuple2;
 import reactor.util.function.Tuples;
@@ -21,7 +20,6 @@ import run.halo.app.core.extension.Menu;
 import run.halo.app.core.extension.MenuItem;
 import run.halo.app.extension.Metadata;
 import run.halo.app.extension.ReactiveExtensionClient;
-import run.halo.app.infra.utils.JsonUtils;
 import run.halo.app.theme.finders.vo.MenuVo;
 
 /**
@@ -44,7 +42,7 @@ class MenuFinderImplTest {
     }
 
     @Test
-    void listAsTree() throws JSONException {
+    void listAsTree() {
         Tuple2<List<Menu>, List<MenuItem>> tuple = testTree();
         Mockito.when(client.list(eq(Menu.class), eq(null), eq(null)))
             .thenReturn(Flux.fromIterable(tuple.getT1()));
@@ -52,141 +50,29 @@ class MenuFinderImplTest {
             .thenReturn(Flux.fromIterable(tuple.getT2()));
 
         List<MenuVo> menuVos = menuFinder.listAsTree();
-        JSONAssert.assertEquals("""
-                [
-                     {
-                         "metadata": {
-                             "name": "D"
-                         },
-                         "spec": {
-                             "displayName": "D",
-                             "menuItems": [
-                                 "E"
-                             ]
-                         },
-                         "menuItems": [
-                             {
-                                 "metadata": {
-                                     "name": "E"
-                                 },
-                                 "spec": {
-                                     "displayName": "E",
-                                     "priority": 0,
-                                     "children": [
-                                         "A",
-                                         "C"
-                                     ]
-                                 },
-                                 "status": {},
-                                 "children": [
-                                     {
-                                         "metadata": {
-                                             "name": "A"
-                                         },
-                                         "spec": {
-                                             "displayName": "A",
-                                             "priority": 0,
-                                             "children": [
-                                                 "B"
-                                             ]
-                                         },
-                                         "status": {},
-                                         "children": [
-                                             {
-                                                 "metadata": {
-                                                     "name": "B"
-                                                 },
-                                                 "spec": {
-                                                     "displayName": "B",
-                                                     "priority": 0
-                                                 },
-                                                 "status": {},
-                                                 "parentName": "A"
-                                             }
-                                         ],
-                                         "parentName": "E"
-                                     },
-                                     {
-                                         "metadata": {
-                                             "name": "C"
-                                         },
-                                         "spec": {
-                                             "displayName": "C",
-                                             "priority": 0
-                                         },
-                                         "status": {},
-                                         "parentName": "E"
-                                     }
-                                 ]
-                             }
-                         ]
-                     },
-                     {
-                         "metadata": {
-                             "name": "X"
-                         },
-                         "spec": {
-                             "displayName": "X",
-                             "menuItems": [
-                                 "G"
-                             ]
-                         },
-                         "menuItems": [
-                             {
-                                 "metadata": {
-                                     "name": "G"
-                                 },
-                                 "spec": {
-                                     "displayName": "G",
-                                     "priority": 0
-                                 },
-                                 "status": {}
-                             }
-                         ]
-                     },
-                     {
-                         "metadata": {
-                             "name": "Y"
-                         },
-                         "spec": {
-                             "displayName": "Y",
-                             "menuItems": [
-                                 "F"
-                             ]
-                         },
-                         "menuItems": [
-                             {
-                                 "metadata": {
-                                     "name": "F"
-                                 },
-                                 "spec": {
-                                     "displayName": "F",
-                                     "priority": 0,
-                                     "children": [
-                                         "H"
-                                     ]
-                                 },
-                                 "status": {},
-                                 "children": [
-                                     {
-                                         "metadata": {
-                                             "name": "H"
-                                         },
-                                         "spec": {
-                                             "displayName": "H",
-                                             "priority": 0
-                                         },
-                                         "status": {},
-                                         "parentName": "F"
-                                     }
-                                 ]
-                             }
-                         ]
-                     }
-                ]
-                """,
-            JsonUtils.objectToJson(menuVos),
-            true);
+        assertThat(visualizeTree(menuVos)).isEqualTo("""
+            D
+            └── E
+                ├── A
+                │   └── B
+                └── C
+            X
+            └── G
+            Y
+            └── F
+                └── H
+            """);
+    }
+
+    /**
+     * Visualize a tree.
+     */
+    String visualizeTree(List<MenuVo> menuVos) {
+        StringBuilder stringBuilder = new StringBuilder();
+        for (MenuVo menuVo : menuVos) {
+            menuVo.print(stringBuilder);
+        }
+        return stringBuilder.toString();
     }
 
     Tuple2<List<Menu>, List<MenuItem>> testTree() {

